### PR TITLE
Add jenkins benchmark job for cluster endpoint runs

### PR DESCRIPTION
**Pull request by Divyaasm**
### Description
Add jenkins benchmark job for cluster endpoint runs

### Issues Resolved
Requires https://github.com/opensearch-project/opensearch-build-libraries/pull/400

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/opensearch/benchmark-test-endpoint.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        )
+        string(
+
+            name: 'TEST_WORKLOAD',
+            description: 'The workload name from OpenSearch Benchmark Workloads.',
+            defaultValue: 'nyc_taxis',
+            trim: true
+        )
+        string(
+            name: 'ADDITIONAL_CONFIG',
```
Comment (rishabh6788) [inline jenkins/opensearch/benchmark-test-endpoint.jenkinsfile:45]: This is related to opensearch-cluster-cdk, not required here.

File: jenkins/opensearch/benchmark-test-endpoint.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        JOB_NAME = 'benchmark-test'
+    }
+parameters {
+        string(
+            name: 'CLUSTER_ENDPOINT',
+            description: 'Provide an endpoint to a cluster for running benchmark tests against it.',
+            trim: true
+        )
+        booleanParam(
+            name: 'CLUSTER_WITH_SECURITY',
```
Comment (rishabh6788) [inline jenkins/opensearch/benchmark-test-endpoint.jenkinsfile:33]: nit: rename to `SECURITY_ENABLED`?

File: tests/jenkins/TestRunBenchmarkTestEndpoint.groovy [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        helper.registerAllowedMethod('findFiles', [Map.class], null)
+        helper.registerAllowedMethod("withCredentials", [Map])
+        binding.setVariable('AGENT_LABEL', 'Jenkins-Agent-AL2-X64-C54xlarge-Docker-Host')
+        binding.setVariable('AGENT_IMAGE', 'opensearchstaging/ci-runner:ci-runner-centos7-v1')
+        binding.setVariable('ARCHITECTURE', 'x64')
+        binding.setVariable('ARTIFACT_BUCKET_NAME', 'test_bucket')
+        binding.setVariable('ARTIFACT_DOWNLOAD_ROLE_NAME', 'Dummy_Download_Role')
+        binding.setVariable('AWS_ACCOUNT_PUBLIC', 'dummy_account')
+        binding.setVariable('CLUSTER_ENDPOINT', 'opensearch-ABCxdfdfhyfk.com')
+        binding.setVariable('CLUSTER_WITH_SECURITY', true)
```
Comment (rishabh6788) [inline tests/jenkins/TestRunBenchmarkTestEndpoint.groovy:54]: Can you please add one more test with security disabled?

Comment (Divyaasm) [reply tests/jenkins/TestRunBenchmarkTestEndpoint.groovy:54]: Sure Rishabh

**Review by rishabh6788 — APPROVED** [review]

**Pull request merged**